### ISSUE TITLE
Swap gets stuck on approving a token for swapping

### ISSUE THREAD

**Issue by asurrogate** [RabbyHub/Rabby#1910]
## TL;DR
I seem to be unable to use the swap function in Rabby because a dialog just closes the whole Rabby modal.
[Screencast from 08.12.2023 01:10:58.webm](https://github.com/RabbyHub/Rabby/assets/153244153/029a4609-fb96-41c8-9f59-6c6d68526f07)

## System

- Ubuntu 0.92.38
- Chrome Version 120.0.6099.71 (Official Build) (64-bit)
- Rabby version 0.92.38
- Using Ethernet main net

### Intended workflow
I want to try swapping tokens inside Rabby.

### What works
- Unlocking Rabby works
- Selecting wallet works
- Selecting token to swap and token to swap into works
- getting quotes works
- selecting one of the quoted sites works

### What does not work
After all the steps under "what works", I am getting stuck on a new alert that features an "Approve <token>" button. 

- Clicking this button collapses the Rabby modal and the Rabby symbol in the browser bar shows a highlighted "1" to symbolize there's a pending task. 
- Clicking on the Rabby symbol in the browser bar brings up an alert that reads "1 transaction needs to sign", options are "View" and "Reject all".
- Clicking the "View" option again collapses the Rabby modal and the Rabby symbol in the browser bar shows a highlighted "1"
- Clicking "Reject all" works, but of course rejects the transaction I wanted to approve

## Workarounds

 I have not found none.

## Attachments

I have attached a video that shows the process and where it fails. Mouse cursor is unfortunately not visible

**Comment by vvvvvv1vvvvvv** [comment]
Does Reject all and re-create the transaction works for you?

**Comment by vvvvvv1vvvvvv** [comment]
No idea for why it happens, did reboot browser or computer works for you?

**Comment by vvvvvv1vvvvvv** [comment]
https://github.com/RabbyHub/Rabby/pull/1916
Trying to fix by this PR

**Comment by asurrogate** [comment]
Thanks, I was able to swap now. Restarting browser or computer did nothing, I tried that. But it seems the merged PR did the trick. 

Thanks for your work, really enjoying Rabby! Happy new year :)
